app/controllers/cla_signatures_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class ClaSignaturesController < ApplicationController
+  before_action :set_repository, set_repository: %i[create]
+
+  def create
```

Comment:
Note: you don't need `@cla_signature` here, since you only redirect.

Also, you explicitly branch on `save!` throw, so it can be simplified like this:

```ruby
message = begin
  authorize ClaSignature.create!(cla_signtaure_params)
  { notice: "CLA has been signed as #{params[:cla_signature][:real_name]}!" }
rescue ActiveRecord::RecordInvalid => error
  { alert: error.message }
end

redirect_to cla_repository_path(@repository.organization.login, @repository.name), message
```

removing all the duplication.